### ISSUE TITLE
NavController remove function breaks the menu back arrow

### ISSUE THREAD

**Issue by mnuckols** [ionic-team/ionic#12312]
**Ionic version:**  (check one with "x")
[ ] **1.x** (For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[x] **3.x**

**I'm submitting a ...**  (check one with "x")
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**
I recently created a new ionic project and now I cannot remove a page from the nav controller using code that worked in previous projects.  When I remove a page lower in the stack, the menu back arrow (in a toolbar) on the current page (top-most) page no longer works.  Clicking it has no effect.

**Expected behavior:**
Calling the nav controller remove(index: number) function should remove the page from the stack and clicking the menu back arrow should pop the current (top-most) page and take you to the next page in the stack.

**Steps to reproduce:**
I'm using the code below to remove a page from the nav controller.  Each page has an ion-navbar on it.

Pages in stack:  A --> B --> Execute Code --> C

With page A & B in the nav stack, execute the code above to show page C.  The code above should remove page B from the stack, so that clicking the menu back arrow will take you to page A.

**Related code:**

```
let curIndex = this.navCtrl.length() - 1;
 this.navCtrl.push('MyNextPage', { user: user }).then(() => {
            this.navCtrl.remove(curIndex);           
});

<ion-navbar color="primary" no-border-bottom>
      <button ion-button menuToggle>
        <ion-icon name="menu"></ion-icon>
      </button> 
</ion-navbar>
```

**Other information:**

**Ionic info:** (run `ionic info` from a terminal/cmd prompt and paste output below):

```
global packages:

    @ionic/cli-utils : 1.4.0
    Cordova CLI      : 6.5.0
    Ionic CLI        : 3.4.0

local packages:

    @ionic/app-scripts              : 2.0.0
    @ionic/cli-plugin-cordova       : 1.4.0
    @ionic/cli-plugin-ionic-angular : 1.3.1
    Cordova Platforms               : android 6.1.2 ios 4.3.1
    Ionic Framework                 : ionic-angular 3.5.0

System:

    Node       : v6.10.0
    OS         : Windows 10
    Xcode      : not installed
    ios-deploy : not installed
    ios-sim    : not installed
    npm        : 3.10.5
```

**Comment by AleksandrChernyavenko** [comment]
I have the same problem. Does anyone know a temporary fix?

**Comment by jgw96** [comment]
Hello @mnuckols , would you mind sharing a repo we could use to reproduce this issue? Thanks for using Ionic!

**Comment by pmurpmur** [comment]
Same issue for me. It is very easy to reproduce, just push 2 views on the stack and remove the first one you pushed - then the back button does not work (no error). It would be nice if Ionic nav had a replace function. I was able to reproduce this in the ionic-conference-app by changing the following method in `SchedulePage`:

```
goToSessionDetail(sessionData: any) {
    // go to the session detail page
    // and pass in the session data

    this.navCtrl.push(SessionDetailPage, { sessionId: sessionData.id, name: sessionData.name });
    this.navCtrl.push(SessionDetailPage, { sessionId: sessionData.id, name: sessionData.name }).then(() => this.navCtrl.remove(0));
  }
```

**Comment by mnuckols** [comment]
I was able to create a temporary workaround using the code below:

```
let curIndex = this.navCtrl.length() - 1;
this.navCtrl.remove(curIndex, 1, { animate: false }).then(() => {
       this.navCtrl.push('NextPage');
});
```

Unfortunately, I can't use this solution in production because it takes longer to transition and shows the previous page briefly before switching to the 'NextPage'.  But at least it's a way to keep my development moving forward for now.

**Comment by peterkelly** [comment]
Possible duplicate of #12238

**Comment by AleksandrChernyavenko** [comment]
Now looks fixed ( version 3.5.3  ).

**Comment by jgw96** [comment]
Hello all! As it looks like this has already been fixed, I am going to close this issue for now. Thanks for using Ionic!
